elk/lightcurve.py

```diff
@@ -646,37 +650,52 @@ def diagnose_lc_periodogram(self, output_path, freq_bins='auto', identifier='',
+            axes[0].scatter(axes[0].get_xlim()[0] + pixel_inds[:, 1] + 0.5,
+                            axes[0].get_ylim()[0] + pixel_inds[:, 0] + 0.5,
+                            c='r', s=1, alpha=0.5)
 
             # create a mask for the frequency range
             frequency_mask = (self.periodogram_freqs >= lower) & (self.periodogram_freqs < upper)
-
+            center = np.median(self.periodogram_freqs[frequency_mask])
             # get power in each pixel that is within the aperture and for the given frequency range
             pixel_powers = aperture_powers[:, frequency_mask]
 
             # get the max power in each pixel
             pixel_max_power = np.zeros([self.cutout_size, self.cutout_size], dtype='float64')
             pixel_max_power[self.star_mask] = np.max(pixel_powers, axis=1)
 
-            if save_simbad_queries:
-                query_pixels = np.argwhere(pixel_max_power > 0.75 * np.max(pixel_max_power))
-                print(query_pixels)
-                ra, dec = self.quality_tpfs.wcs.pixel_to_world_values(*query_pixels.T)
+            if query_simbad:
+                query_pixels = np.argwhere(pixel_max_power > 0.9 * np.max(pixel_max_power))
+
+                ra, dec = np.array(self.quality_tpfs.wcs.pixel_to_world_values(np.fliplr(query_pixels)))[:,0], np.array(self.quality_tpfs.wcs.pixel_to_world_values(np.fliplr(query_pixels)))[:,1] 
+
                 simbad_queries['ra'] = np.concatenate((simbad_queries['ra'], ra))
                 simbad_queries['dec'] = np.concatenate((simbad_queries['dec'], dec))
-                simbad_queries['freq_lower'] = np.concatenate((simbad_queries['freq_lower'],
-                                                               np.repeat(lower, len(ra))))
-                simbad_queries['freq_upper'] = np.concatenate((simbad_queries['freq_upper'],
-                                                               np.repeat(upper, len(ra))))
-
-            # plot the max power in each pixel in the same range as the left panel
+                simbad_queries['peak_freq'] = np.concatenate((simbad_queries['peak_freq'],
+                                                               np.repeat(peak_center, len(ra))))
+                
+                # create custom simbad that includes variable star columns
+                var_Simbad = Simbad()
+                var_Simbad.add_votable_fields('v*', 'otype', 'flux(V)')
+
+                query_result= var_Simbad.query_region(coord.SkyCoord(ra=ra, dec=dec,
+                                                      unit=(u.deg, u.deg), frame='icrs'),
+                                                      radius=0.008 * u.deg)
+                
+                
+                
+
+                if query_result is not None:
+                    query_result.add_column([round(center, 2)], name='Peak_Frequency')
+                else: query_result= 'No Simbad Objects in Pixel'
+
+                print(query_result)
```

Comment:
Should probably only print when `verbose`, I figure you'd want to save this table instead? Also the statement after the else above should usually be on a new line

Reply:
Agreed!